### PR TITLE
First working version

### PR DESCRIPTION
**Pull request by dignifiedquire**
Depends on https://github.com/ipfs/js-libp2p-ipfs/pull/6

Closes https://github.com/ipfs/js-ipfs/issues/126

### CODE REVIEW

File: gulpfile.js [hunk 1]
```diff
@@ -0,0 +1,33 @@
...
+
+const gulp = require('gulp')
+const multiaddr = require('multiaddr')
+const Node = require('./src').Node
+const Peer = require('peer-info')
+
+let node
+
+gulp.task('libnode:start', (done) => {
+  node = new Node()
```
Comment (diasdavid) [inline gulpfile.js]: You forgot to call `node.start(cb)`

File: gulpfile.js [hunk 2]
```diff
@@ -0,0 +1,33 @@
...
+  const mh = multiaddr('/ip4/127.0.0.1/tcp/9090/websockets')
+  const peer = new Peer()
+  peer.multiaddr.add(mh)
+
+  node.swarm.handle(peer, '/echo/1.0.0', (conn) => {
```
Comment (diasdavid) [inline gulpfile.js]: This call doesn't take any peerInfo object https://github.com/diasdavid/js-libp2p-swarm/blob/master/src/index.js#L304

File: gulpfile.js [hunk 3]
```diff
@@ -0,0 +1,36 @@
+'use strict'
+
+const gulp = require('gulp')
+const multiaddr = require('multiaddr')
+const Node = require('./src').Node
```
Comment (diasdavid) [inline gulpfile.js]: you are using the same libp2p-ipfs-browser to fire the Node.js node (which does nothing on swarm)

Comment (dignifiedquire) [inline gulpfile.js]: -.- right thanks

File: gulpfile.js [hunk 4]
```diff
@@ -0,0 +1,34 @@
...
+const Peer = require('peer-info')
+
+let node
+
+gulp.task('libnode:start', (done) => {
+  const mh = multiaddr('/ip4/127.0.0.1/tcp/9090/websockets')
+  const peer = new Peer()
+
+  node = new Node(peer)
+  node.start(() => {})
```
Comment (diasdavid) [inline gulpfile.js]: This means that the gulp script finishes before the socket being open

**Comment by diasdavid** [comment]
Greeeeeeeen :D

**Pull request merged**